### PR TITLE
remove CTC targets from Datasets

### LINKED ISSUE #1179: Remove CTC Targets from all Datasets
From my understanding this very specific feature was only used in combination with Theano setups. If yes, it can be removed.

### PR DESCRIPTION
**Pull request by JackTemaki**
resolves #1179

### CODE REVIEW

File: returnn/datasets/basic.py [hunk 1]
```diff
@@ -893,20 +877,6 @@ def serialize_data(self, key, data):
...
-    :rtype: numpy.ndarray
-    """
-    priori = numpy.zeros((self.num_outputs[target][0],), dtype=numpy.float32)
-    i = 0
-    while self.is_less_than_num_seqs(i):
-      self.load_seqs(i, i + 1)
-      for t in self.get_targets(target, i):
-        priori[t] += 1
-      i += 1
-    return numpy.array(priori / self.get_num_timesteps(), dtype=numpy.float32)
```
Comment (albertz) [inline returnn/datasets/basic.py:908]: This is somewhat unrelated to this PR. But I guess also ok.

Comment (JackTemaki) [reply returnn/datasets/basic.py:908]: Ah yes, `update_weights` is extra as well. I forgot to exclude it from the commit.

**Review by albertz — APPROVED** [review]
